app/src/main/java/app/nexd/android/ui/seeker/create/SeekerCreateRequestViewModel.kt

```diff
@@ -1,47 +1,78 @@
 package app.nexd.android.ui.seeker.create
 
+import androidx.annotation.StringRes
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.LiveDataReactiveStreams
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import app.nexd.android.Api
+import app.nexd.android.R
 import app.nexd.android.api.model.Article
+import app.nexd.android.api.model.CreateHelpRequestArticleDto
 import app.nexd.android.api.model.HelpRequestCreateDto
 import app.nexd.android.api.model.User
+import app.nexd.android.ui.common.HelpRequestCreateArticleBinder
 import io.reactivex.BackpressureStrategy
-import io.reactivex.processors.BehaviorProcessor
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
 
 class SeekerCreateRequestViewModel(private val api: Api) : ViewModel() {
 
-    enum class State {
-        LOADING,
-        IDLE,
-        FINISHED
+    sealed class Progress {
+        object Idle : Progress()
+        class Error(@StringRes val message: Int) : Progress()
+        object Finished : Progress()
     }
 
-    private val state = BehaviorProcessor.createDefault(State.LOADING)
+    val progress = MutableLiveData<Progress>(Progress.Idle)
 
-    fun getCurrentUser(): LiveData<User> {
-        return LiveDataReactiveStreams.fromPublisher(
-            api.userControllerFindMe().toFlowable(BackpressureStrategy.LATEST)
-        )
-    }
+    val articles = LiveDataReactiveStreams.fromPublisher(
+        api.articlesControllerFindAll()
+            .map { articles ->
+                articles.map { HelpRequestCreateArticleBinder.ArticleInput(it) }
+            }
+            .toFlowable(BackpressureStrategy.BUFFER)
+    )
 
-    fun getArticles() : LiveData<List<Article>> {
-        return LiveDataReactiveStreams.fromPublisher(
-            api.articlesControllerFindAll().toFlowable(BackpressureStrategy.BUFFER)
-        )
-    }
+    private val compositeDisposable = CompositeDisposable()
 
-    fun sendRequest(request: HelpRequestCreateDto) {
-        if (request.articles.isNullOrEmpty()) {
-            return // request without articles shouldn't be accepted
-        }
-        with(api) {
-            helpRequestsControllerInsertRequestWithArticles(request)
-                .subscribe { state.onNext(State.FINISHED) }
-        }
-    }
+    fun sendRequest() {
+        val articles = this.articles.value
+            ?.filter { it.amount > 0 }
+            ?.map {
+                CreateHelpRequestArticleDto()
+                    .articleCount(it.amount)
+                    .articleId(it.article.id)
+            } ?: kotlin.run {
+                return
+            }
 
-    fun state() = LiveDataReactiveStreams.fromPublisher(state)
```

Comment:
would be interessting for me to understand when `LiveDataReactiveStreams` should be used.
Depending on if we want to handle onError ?

Reply:
i answered this above:)